classes/base.lua

```diff
@@ -77,21 +84,28 @@ function class:_post_init ()
   SILE.scratch.half_initialized_class = nil
 end
 
-function class:setOptions (options)
+function class:setOptions(options)
   options = options or {}
+  options.orientation = options.orientation or "portrait"
+
+  if not self.options.orientation then
+    self.options.orientation = options.orientation
```

Comment:
Should not be necessary (you added it to `options` on line 89, and lines 97-99 therefore propagate it to the class instance already, methink.

Reply:
As I explained in the first commit, while testing this feature, sometimes it suddenly stopped to work and that is the only way I got to solve the problem.

> After some time I found out why: since there is no order in a loop using
> pairs(), sometimes the loop in class:setOptions() was getting orientation
> as the last element, so when SILE.papersize was called, the value of
> options.orientation was indeed nil, so the solution should be just
> set the orientation value to class.options before papersize.

But indeed, it looks a bit redundant and could be more straight...

Reply:
That suggests there is some sort of race condition here, probably the order in which options are handled. Papersize and orientation might be handled in the opposite order (table `pairs()` iteration is non-deterministic).

Reply:
@jodros Please don't mark review comments as resolved if you didn't start them. Leave it to the people that started the thread or a maintainer to determine whether they are resolved.